pkg/gitops/flux/flux.go

```diff
@@ -218,7 +219,7 @@ func (f *Flux) UpdateGitEksaSpec(ctx context.Context, clusterSpec *cluster.Spec,
 		return nil
 	}
 
-	fc := newFluxForCluster(f, clusterSpec, datacenterConfig, machineConfigs)
+	fc := newFluxForCluster(f, clusterSpec, datacenterConfig, machineConfigs, nil)
```

Comment:
I don't love passing nil. Is there anyway we can avoid it?
Can it just be an argument to the method that needs it?